### ISSUE TITLE
torch YatConv2D: DropConnect uses boolean mask without keep-probability rescaling

### ISSUE THREAD

**Issue by mlnomadpy** [azettaai/nmn#21]
## Summary

`YatConv2D` is the only YAT torch conv layer whose DropConnect path uses a boolean mask and *forgets to divide by the keep probability*. The other five conv variants (`YatConv1D`, `YatConv3D`, `YatConvTranspose1D/2D/3D`) all use the canonical inverted-dropout pattern: `bernoulli(keep_prob) * weight / keep_prob`. The `YatConv2D` form silently scales the effective weight magnitude by `(1 − drop_rate)` during training, biasing both forward activations and gradients downward.

## Affected lines

`src/nmn/torch/layers/yat_conv2d.py:252-256`:

```python
if self.use_dropconnect and not deterministic and self.drop_rate > 0.0:
    if self.training:
        dropout_mask = torch.rand_like(weight) > self.drop_rate
        weight = weight * dropout_mask
```

Compare with the correct pattern in `yat_conv1d.py:241-245`:

```python
if self.use_dropconnect and not deterministic and self.drop_rate > 0.0:
    if self.training:
        keep_prob = 1.0 - self.drop_rate
        drop_mask = torch.bernoulli(torch.full_like(weight, keep_prob))
        weight = (weight * drop_mask) / keep_prob
```

Identical pattern (correct) appears in:
- `yat_conv3d.py:238-242`
- `yat_conv_transpose1d.py:182-185`
- `yat_conv_transpose2d.py:182-185`
- `yat_conv_transpose3d.py:182-185`

## Why it's a bug

- Train/eval expectation mismatch: with `drop_rate=0.5`, training-time effective weights have ~50% of expected magnitude, but inference uses the full weight → distribution shift at eval time, exactly what inverted dropout exists to avoid.
- YAT is doubly sensitive because the numerator is squared: a 0.5× rescale of weight → 0.25× rescale of `(x·W + b)²`, while the distance denominator also changes — gradient magnitudes get distorted nonlinearly.
- Inconsistent with the other five layers in the same package, which makes the bug easy to miss.

## Suggested fix

Match the YatConv1D pattern:

```python
if self.use_dropconnect and not deterministic and self.drop_rate > 0.0:
    if self.training:
        keep_prob = 1.0 - self.drop_rate
        drop_mask = torch.bernoulli(torch.full_like(weight, keep_prob))
        weight = (weight * drop_mask) / keep_prob
```

## Repro

```python
import torch
from nmn.torch.layers.yat_conv1d import YatConv1D
from nmn.torch.layers.yat_conv2d import YatConv2D

torch.manual_seed(0)
x = torch.randn(64, 8, 16, 16)
l2 = YatConv2D(8, 16, 3, use_dropconnect=True, drop_rate=0.5).train()
print("YatConv2D train mean magnitude:", l2(x).abs().mean().item())
l2.eval()
print("YatConv2D eval  mean magnitude:", l2(x).abs().mean().item())
# expect ~equal; observe train << eval
```
